### ISSUE TITLE
windows error in diagnostics tests

### ISSUE THREAD

**Issue by cowlicks** [blaze/dask#751]
```python
_________________________ test_resource_profiler_plot _________________________

    @pytest.mark.skipif("not bokeh")
    @pytest.mark.skipif("not psutil")
    def test_resource_profiler_plot():
        with ResourceProfiler(dt=0.01) as rprof:
            get(dsk2, 'c')
        p = rprof.visualize(plot_width=500,
                            plot_height=300,
                            tools="hover",
                            title="Not the default",
>                           show=False, save=False)

tests\test_profiler.py:160:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
profile.py:169: in visualize
    return visualize(self, **kwargs)
profile_visualize.py:170: in visualize
    figs = [prof._plot(**kwargs) for prof in profilers]
profile.py:159: in _plot
    return plot_resources(self.results, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

results = [], palette = 'GnBu'
kwargs = {'plot_height': 300, 'plot_width': 500, 'title': 'Not the default', 'tools': 'hover'}

    def plot_resources(results, palette='GnBu', **kwargs):
        """Plot resource usage in a bokeh plot.

        Parameters
        ----------
        results : sequence
            Output of ResourceProfiler.results
        palette : string, optional
            Name of the bokeh palette to use, must be key in bokeh.palettes.brewer.
        **kwargs
            Other keyword arguments, passed to bokeh.figure. These will override
            all defaults set by plot_resources.

        Returns
        -------
        The completed bokeh plot object.
        """
>       t, mem, cpu = zip(*results)
E       ValueError: need more than 0 values to unpack

profile_visualize.py:287: ValueError
_______________________________ test_plot_both ________________________________

    @pytest.mark.skipif("not bokeh")
    @pytest.mark.skipif("not psutil")
    def test_plot_both():
        from dask.diagnostics.profile_visualize import visualize
        from bokeh.plotting import GridPlot
        with ResourceProfiler(dt=0.01) as rprof:
            with prof:
                get(dsk2, 'c')
        p = visualize([prof, rprof], label_size=50,
>                     title="Not the default", show=False, save=False)

tests\test_profiler.py:177:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
profile_visualize.py:170: in visualize
    figs = [prof._plot(**kwargs) for prof in profilers]
profile.py:159: in _plot
    return plot_resources(self.results, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

results = [], palette = 'GnBu'
kwargs = {'label_size': 50, 'title': 'Not the default'}

    def plot_resources(results, palette='GnBu', **kwargs):
        """Plot resource usage in a bokeh plot.

        Parameters
        ----------
        results : sequence
            Output of ResourceProfiler.results
        palette : string, optional
            Name of the bokeh palette to use, must be key in bokeh.palettes.brewer.
        **kwargs
            Other keyword arguments, passed to bokeh.figure. These will override
            all defaults set by plot_resources.

        Returns
        -------
        The completed bokeh plot object.
        """
>       t, mem, cpu = zip(*results)
E       ValueError: need more than 0 values to unpack

profile_visualize.py:287: ValueError
=============== 2 failed, 17 passed, 1 skipped in 47.16 seconds ===============
PS C:\Users\IEUser\Documents\GitHub\dask\dask\diagnostics>
```

**Comment by cowlicks** [comment]
cc @jcrist should this work on windows?

**Comment by jcrist** [comment]
This looks like an intermittent error - the results list is empty. If there are never any results (say for a longer test) then it might be a windows issue. Either way, that `zip` should be changed to `unzip` from `dask.imperative` (and that function should be moved to `utils`)

**Comment by cowlicks** [comment]
This happens consistently.

**Comment by jcrist** [comment]
Blake and I tested this on a windows VM, and found it works for longer runtimes - the test is too short for windows to spin up a process, log data, then send it. My vote is to leave the test as is (unix-likes test that data is recorded), and the fix in #754 ensures that plotting works with empty results. Other option is to make the test longer. Thoughts?

**Comment by jcrist** [comment]
Tested on windows, closing.
